extensions/wikia/ContributionPrototype/CPArticleRenderer.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace ContributionPrototype;
+
+use Http;
+use MWHttpRequest;
+use OutputPage;
+use Wikia;
+use Wikia\Logger\Loggable;
+use Wikia\Service\Gateway\UrlProvider;
+
+class CPArticleRenderer {
+	
+	use Loggable;
+
+	const SERVICE_NAME = "structdata";
+
+	/** @var string */
+	private $publicHost;
+	
+	/** @var int */
+	private $wikiId;
+	
+	/** @var string */
+	private $dbName;
+
+	/** @var UrlProvider */
+	private $urlProvider;
+
+	/**
+	 * CPArticleRenderer constructor.
+	 * @param string $publicHost
+	 * @param int $wikiId
+	 * @param string $dbName
+	 * @param UrlProvider $urlProvider
+	 */
+	public function __construct($publicHost, $wikiId, $dbName, $urlProvider) {
+		$this->publicHost = $publicHost;
+		$this->wikiId = $wikiId;
+		$this->dbName = $dbName;
+		$this->urlProvider = $urlProvider;
+	}
+
+	/**
+	 * @param string $title
+	 * @param OutputPage $output
+	 */
+	public function render($title, OutputPage $output) {
+		$content = $this->getArticleContent($title);
+		
+		if ($content === false) {
+			// TODO: what do we want to show here?
+			return;
+		}
+
+		$this->addStyles($output);
+		$this->addScripts($output);
+		$output->addHTML($content);
+	}
+
+	private function addStyles(OutputPage $output) {
+		$output->addLink([
+				'rel' => 'stylesheet',
+				'href'=> "{$this->publicHost}/public/assets/styles/main.css",
+		]);
+
+		// this ends up using $wgOut, but we need it for the assets manager integration, no point in duplicating :(
+		Wikia::addAssetsToOutput('contribution_prototype_scss');
+	}
+
+	private function addScripts(OutputPage $output) {
+		$output->addScript("<script src=\"{$this->publicHost}/public/assets/app.js\"></script>");
+	}
+
+	private function getArticleContent($title) {
+//		$internalHost = $this->urlProvider->getUrl(self::SERVICE_NAME);
```

Comment:
I intentionally left this in, because we do want to use the `urlProvider` here to get an internal url. For now there's only one instance of the CP in dev, so we can't use the url provider (since that's environment aware). When we launch a prod instance we will change this to use the internal url provided by the `UrlProvider`